keps/sig-auth/3221-multiple-authorization-webhooks/README.md

```diff
@@ -0,0 +1,479 @@
+
+#### Beta
+
+- Address user reviews and iterate (if needed keep in Alpha until changes stabilize)
+- Feature flag will be turned on by default
+
+#### GA
+
+- Feature flag removed
+- Existing command line flags will have no effect
+
+#### GA + 3 cycles
+
+- Remove the existing command line flags
+
+### Upgrade / Downgrade Strategy
+
+While the feature is in Alpha, there is no change if cluster administrators want to keep on using command line flags.
+
+When the feature goes to Beta/GA or the cluster administrators want to configure authorizers using a config file, they need to make sure the config file exists before upgrading the cluster. Similarly when downgrading clusters, they would need to add the flags back to their bootstrap mechanism.
+
+### Version Skew Strategy
+
+Not applicable.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate
+  - Feature gate name: `AuthorizationConfigFromFile`
+  - Components depending on the feature gate:
+        - kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Yes, kube-apiserver will not use the `--authorization-*` to configure the authorizer chain. `kube-apiserver` will need the `--authorization-config-file` flag to be set to a valid config file on disk.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled once enabled. If the `--authorization-*` flags are not removed while enabling the feature, past behavior would be restored.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+###### Are there any tests for feature enablement/disablement?
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout can fail when the authorization configuration file being passed doesn't exist or is invalid.
+
+Already running workloads won't be impacted but cluster users won't be able to access the control plane if the cluster is single-node.
+
+###### What specific metrics should inform a rollback?
+
+Not applicable.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+TBD.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+None until the feature reaches GA and at a future point in time, `--authorization-*` flags other than `--authorization-config-file` are removed.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Not applicable.
```

Comment:
this would be checking to see if the flag is set, correct?

Reply:
Yes, cluster administrators can view the kube-apiserver configuration to know if an AuthorizationConfiguration file is being used.

On that note, we can also add a log entry when we detect a configuration file is passed.